src/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import p from 'path'
+import requireFromStringOfCode from 'require-from-string'
+import type babelCore from '@babel/core'
+import objectToAST from './object-to-ast'
+
+type PrevalModuleExport = unknown | ((...args: Array<any>) => unknown)
+
+type CompiledESModule = {
+  __esModule: boolean
+  default: PrevalModuleExport
+}
+
+function isCompiledESModule(module: unknown): module is CompiledESModule {
+  return typeof module === 'object' && module !== null && '__esModule' in module
+}
+
+// istanbul ignore next because I don't know how to reproduce a situation
+// where the filename doesn't exist, but TypeScript gets mad when I don't handle that case.
+const getFilename = (fileOpts: babelCore.TransformOptions): string =>
+  fileOpts.filename ?? '"unknown"'
+
+type RequireFromStringOptions = {
+  string: string
+  fileOpts: babelCore.TransformOptions
+  args?: any[]
+}
+export function requireFromString({
+  string: stringToPreval,
+  fileOpts,
+  args = [],
+}: RequireFromStringOptions) {
+  const filename = getFilename(fileOpts)
+  let module = requireFromStringOfCode(String(stringToPreval), filename) as
+    | CompiledESModule
+    | PrevalModuleExport
+    | unknown
+
+  if (isCompiledESModule(module)) {
+    // Allow for es modules (default export)
+    module = module.default
+  }
```

Comment:
I made a small change here to help satisfy the TS hints.  I believe it's functionally the equivalent (and all the tests still pass), but please review with some extra care.